### PR TITLE
Loss class

### PR DESCRIPTION
**Pull request by campsd**
Class for loss functions.

Part of #54, #55 and required for #70.

### CODE REVIEW

File: funfact/loss.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import jax.numpy as np
```
Comment (yhtang) [inline funfact/loss.py:3]: Now we can replace line with the backend import. 😎

File: funfact/loss.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class Loss:
```
Comment (yhtang) [inline funfact/loss.py:6]: I think it'll be better to actually split this to three classes, one for each loss function. This one currently looks like a factory method. Do you have some concern regarding separating them?

**Review by yhtang — CHANGES_REQUESTED** [review]
If you are worried about duplicating code, e.g.  for the data dim checks, that can be done by factoring out the function as a private method (aka `_` prefixed).

File: funfact/loss.py (hunk 1 above)
Comment (campsd) [reply funfact/loss.py]: Done.

File: funfact/loss.py (hunk 2 above)
Comment (campsd) [reply funfact/loss.py]: I have split it into 3 classes. If there are loss functions that differ from the default behavior, they can override the `__call__` function from the base class.

**Review by yhtang — APPROVED** [review]
